### PR TITLE
prevent _make_arma_names from adding exog_names when already present

### LINKED ISSUE #1691: BUG: ARIMA exog_names expands in each fit.
trying out the example from test_arima. small sample

If I call fit repeatedly, then summary() raises an exception.

It looks like `exog_names` is appended in each call of fit.

```
    y = [-1214.360173, -1848.209905, -2100.918158, -3647.483678, -4711.186773]
    mod = ARIMA(y, (1, 0, 2))

>>> mod.exog_names
['const', 'const', 'const', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y']

>>> res = mod.fit(disp=0, start_params=[np.mean(y), .1, .1, .1])
>>> mod.exog_names
['const', 'const', 'const', 'const', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y']

>>> res = mod.fit(disp=0, start_params=[np.mean(y), .1, .1, .1])
>>> mod.exog_names
['const', 'const', 'const', 'const', 'const', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y']
```

working in a plain Windows shell in py3.3 virtualenv

### PR DESCRIPTION
**Pull request by kiyoto**
closes #1691

I have some doubts about my approach here. It looks like exog_names was added to address Issue #626 back in the day, but it almost seems to assume that `_make_arma_names` (and hence `fit`) is run at most once against the model.

### CODE REVIEW

**Comment by josef-pkt** [comment]
We still have the trend choice as argument in fit.
If users calls fit a second time with different trend, then this will not work, AFAICS.

It will work for repeated calls with identical keywords

**Comment by josef-pkt** [comment]
I think having `trend` as `fit` argument is still a mistake. We cannot have two results instances with different trend option with the same model instance, I guess. 

If we remove trend from `fit` arguments, then there is no need anymore to create the parameter names repreatedly.

I think we should merge this for 0.7 to fix the bug for the main usecase when we call fit repeatedly with different optimizer options and starting values. This case looks properly covered by this PR.

File: statsmodels/tsa/tests/test_arima.py [hunk 1]
```diff
@@ -2210,6 +2210,13 @@ def test_ARIMA_exog_predict():
     assert_allclose(dpredict_002, res_d002[-len(dpredict_002):], rtol=1e-4, atol=1e-6)
 
 
+def test_arima_fit_mutliple_calls():
+    y = [-1214.360173, -1848.209905, -2100.918158, -3647.483678, -4711.186773]
+    mod = ARIMA(y, (1, 0, 2))
+    # Make multiple calls to fit
+    mod.fit(disp=0, start_params=[np.mean(y), .1, .1, .1])
+    mod.fit(disp=0, start_params=[np.mean(y), .1, .1, .1])
+    assert_equal(mod.exog_names,  ['const', 'ar.L1.y', 'ma.L1.y', 'ma.L2.y'])
```
Comment (josef-pkt) [inline statsmodels/tsa/tests/test_arima.py]: to make the test stronger, you can copy the same assert_equal also after the first call to `fit`

**Comment by kiyoto** [comment]
@josef-pkt Thanks.

>If users calls fit a second time with different trend, then this will not work, AFAICS.

You are right. It's possible to handle, but it means leaking the logic of `vector_ar/util.py:make_lag_names` into `_make_arma_names` (also, I don't think trend_name should be calculated inside `make_lag_names`). All that said, it should be a separate conversation.

**Comment by josef-pkt** [comment]
I'm merging this as the first step to the more general fix.

**Pull request merged**